semantic-ast/src/AST/GenerateSyntax.hs

```diff
@@ -62,76 +66,80 @@ getAllSymbols language = do
       let named = if t == 0 then Named else Anonymous
       pure (n, named)
 
+shapeParameterName, annParameterName :: Name
+shapeParameterName = mkName "f"
+annParameterName = mkName "a"
+
 -- Auto-generate Haskell datatypes for sums, products and leaf types
 syntaxDatatype :: Ptr TS.Language -> [(String, Named)] -> Datatype -> Q [Dec]
 syntaxDatatype language allSymbols datatype = skipDefined $ do
-  typeParameterName <- newName "a"
+  shapeFunctorKind <- [t| * -> * |]
+  let traversalInstances = mappend <$> makeStandaloneDerivings (conT name) <*> makeTraversalInstances (conT name)
+      glue a b c = a : b <> c
+      name = mkName nameStr
+      generatedDatatype cons = dataD (cxt []) name [kindedTV shapeParameterName shapeFunctorKind, plainTV annParameterName] Nothing cons [deriveStockClause, deriveAnyClassClause]
```

Comment:
We generate `f :: * -> *` parameters for the record types, in TH, and add them to the generated datatype.